DashAI/back/plugins/utils.py

```diff
@@ -147,12 +116,91 @@ def install_plugin_from_pypi(pypi_plugin_name: str) -> None:
     RuntimeError
         If pip install command fails
     """
+    pip_action = "install" if install else "uninstall"
+    args = ["pip", pip_action, pypi_plugin_name]
+    args if install else args.append("-y")
+
     res = subprocess.run(
-        ["pip", "install", pypi_plugin_name],
+        args,
         stderr=subprocess.PIPE,
         text=True,
     )
     if res.returncode != 0:
         errors = [line for line in res.stderr.split("\n") if "ERROR" in line]
         error_string = "\n".join(errors)
         raise RuntimeError(error_string)
+
+
+def install_plugin(plugin_name: str) -> List[type]:
+    """
+    Install and register new plugins in component registry
+
+    Parameters
+    ----------
+    plugin_name : str
+        A string with the name of the plugin in pypi to install
+
+    component_registry : ComponentRegistry
+        The current app component registry
+
+    """
+    pre_installed_plugins: List[type] = get_available_plugins()
+    install_plugin_from_pypi(plugin_name)
+    installed_plugins = set(get_available_plugins()) - set(pre_installed_plugins)
+    return installed_plugins
+
+
+def register_plugin_components(installed_plugins, component_registry):
+    for plugin in installed_plugins:
+        try:
+            component_registry.register_component(plugin)
+        except Exception as e:
+            logging.exception(f"Plugin {plugin} could not be registered. Error: {e}")
+
+
+def uninstall_plugin(
+    plugin_name: str,
+) -> List[type]:
+    """
+    Uninstall an existing plugin and delete it from component registry
+
+    Parameters
+    ----------
+    plugin_name : str
+        A string with the name of the plugin in pypi to install
+
+    component_registry : ComponentRegistry
+        The current app component registry
+
+    """
+    available_plugins: List[type] = get_available_plugins()
+    install_plugin_from_pypi(plugin_name, False)
+    uninstalled_components: List[type] = set(available_plugins) - set(
+        get_available_plugins()
+    )
+    return uninstalled_components
+
+
+def unregister_plugin_components(
+    uninstalled_plugins: List[type],
+    component_registry: ComponentRegistry,
+) -> List[type]:
+    """
+    Remove from component registry uninstalled plugins
+
+    Parameters
+    ----------
+    component_registry : ComponentRegistry
+        The current app component registry
+
+    Returns
+    ----------
+    List[type]
+        A list of plugins' classes wanted to be removed from the component registry
+    """
+    for plugin in uninstalled_plugins:
+        try:
+            component_registry.unregister_component(plugin)
+        except Exception as e:
+            logging.exception(e)
```

Comment:
Use a file specific logger, not the global one:

```suggestion
        except Exception as e:
            logger.exception(e)
```

And in the top of the file:
```python
logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)
```

Reply:
Due to the change in the code:
Old:
```
try:
     component_registry.register_component(plugin)
except Exception as e:
     logging.exception(f"Plugin {plugin} could not be registered. Error: {e}")
```
Now:
```
component_registry.register_component(plugin)
```
This comment is no longer relevant, because we are no longer using the logging.